api/tests/authentication_tests.py

```diff
@@ -370,3 +370,51 @@ def test_200_resp_from_fxa_for_user_returns_user_and_caches(self):
         # now check that the 2nd call did NOT make another fxa request
         assert responses.assert_call_count(self.fxa_verify_path, 1) is True
         assert cache.get(get_cache_key(user_token)) == fxa_response
+
+    @responses.activate()
+    def test_write_requests_make_calls_to_fxa(self):
+        self.sa = baker.make(SocialAccount, uid=self.uid, provider="fxa")
+        user_token = "user-123"
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=f"Bearer {user_token}")
+        now_time = int(datetime.now().timestamp())
+        # Note: FXA iat and exp are timestamps in *milliseconds*
+        exp_time = (now_time + 60 * 60) * 1000
+        fxa_response = _setup_fxa_response(
+            200, {"active": True, "sub": self.uid, "exp": exp_time}
+        )
+
+        assert cache.get(get_cache_key(user_token)) is None
+
+        # check the endpoint status code
+        response = client.get("/api/v1/relayaddresses/")
+        assert response.status_code == 200
+        assert responses.assert_call_count(self.fxa_verify_path, 1) is True
+        assert cache.get(get_cache_key(user_token)) == fxa_response
+
+        # check the function returns the right user
+        headers = {"HTTP_AUTHORIZATION": f"Bearer {user_token}"}
+        get_addresses_req = self.factory.get(self.path, **headers)
+        auth_return = self.auth.authenticate(self.auth, get_addresses_req)
+        assert auth_return == (self.sa.user, user_token)
+
+        # now check that the 2nd GET request did NOT make another fxa request
+        assert responses.assert_call_count(self.fxa_verify_path, 1) is True
+        assert cache.get(get_cache_key(user_token)) == fxa_response
+
+        headers = {"HTTP_AUTHORIZATION": f"Bearer {user_token}"}
+
+        # send POST to /api/v1/relayaddresses and check that cache is used - i.e.,
+        # FXA is *NOT* called
+        post_addresses_req = self.factory.post(self.path, **headers)
+        auth_return = self.auth.authenticate(self.auth, post_addresses_req)
+        assert responses.assert_call_count(self.fxa_verify_path, 1) is True
```

Comment:
Suggestion (blocking): Another test with `POST` but the cache is not used.

Reply:
As in another test to different endpoint where the POST request should not use the cache.

Reply:
oh oh ... *ADD* another test. Will do.

Reply:
Updated to test that a `POST` to another API endpoint **WILL** cause another FXA token introspection request.